libs/SalesforceSDKCore/SalesforceSDKCore/Classes/NativeLoginManager.swift

```diff
@@ -71,4 +71,66 @@ public protocol NativeLoginManager {
     ///
     /// Note: this call will dismiss your login view controller.
     @objc func biometricAuthenticationSuccess()
+    
+    // MARK: Headless, Password-Less Login Via One-Time-Passcode
+    
+    /// Submits a request for a one-time-passcode to the Salesforce headless password-less login flow.
+    /// This fulfills step three of the headless password-less login flow.
+    ///
+    /// See https://help.salesforce.com/s/articleView?id=sf.remoteaccess_headless_passwordless_login_public_clients.htm&type=5
+    ///
+    /// - Parameters:
+    ///   - username: A valid Salesforce username.  Note that email may be used for community users
+    ///   - recaptchaToken: A reCAPTCHA token provided by the reCAPTCHA SDK
+    ///   - otpVerificationMethod: The delivery method for the OTP
+    /// - Returns: An OTP request result with the overall login result and the OTP identifier for
+    /// successful OTP requests
+    ///
+    @objc func submitOtpRequest(
+        username: String,
+        reCaptchaToken: String,
+        otpVerificationMethod: OtpVerificationMethod) async throws -> OtpRequestResult
+    
+    /// Submits a request for a one-time-passcode to the Salesforce headless password-less login flow.
+    /// This fulfills steps eight, eleven and thirteen of the headless password-less login flow.
+    ///
+    /// See https://help.salesforce.com/s/articleView?id=sf.remoteaccess_headless_passwordless_login_public_clients.htm&type=5
+    ///
+    /// - Parameters:
+    ///   - otp: A user-entered OTP
+    ///   - otpIdentifier: The OTP identifier issued by the Headless Identity API
+    ///   - otpVerificationMethod: The OTP verification method used to obtain the OTP identifier
+    /// - Returns: A login result indicating the outcome of the authorization and access token requests
+    ///
+    @objc func submitPasswordlessAuthorizationRequest(
+        otp: String,
+        otpIdentifier: String,
+        otpVerificationMethod: OtpVerificationMethod
+    ) async throws -> NativeLoginResult
+}
+
+/// An Objective-C compatible OTP request result
+@objc(SFOtpRequestResult)
+@objcMembers
+public class OtpRequestResult: NSObject {
+    
+    /// The overall result of the OTP request.
+    public let nativeLoginResult: NativeLoginResult
+    
+    /// On success result, the OTP identifier provided by the API
+    public let otpIdentifier: String?
+    
+    init(
+        nativeLoginResult: NativeLoginResult,
+        otpIdentifier: String? = nil
+    ) {
+        self.nativeLoginResult = nativeLoginResult
+        self.otpIdentifier = otpIdentifier
+    }
+}
+
+/// The possible OTP verification methods.
+@objc public enum OtpVerificationMethod: Int {
```

Comment:
Should this and `OtpRequestResult` be up at the top of the file with `NativeLoginResult`?

Reply:
It could be.  I always look for the type the source is named for at the top of a given file, so that was my style precedent.

Reply:
Fair enough.  I don't really care if they are all at the top or bottom of the file as long as it is consistent.